mr/src/main/java/org/apache/iceberg/mr/InputFormatConfig.java

```diff
@@ -85,6 +88,13 @@ public ConfigBuilder(Configuration conf) {
     }
 
     public Configuration conf() {
+      // Store the io and the current snapshot of the table in the configuration which are needed for the split
+      // generation
+      Table table = Catalogs.loadTable(conf);
+      conf.set(InputFormatConfig.METADATA_LOCATION,
+          ((HasTableOperations) table).operations().current().metadataFileLocation());
```

Comment:
As I noted on a thread below, I think it is better to use `BaseTable`.

Reply:
Using Serialization